Generator/Sources/NeedleFramework/Parsing/FileFilters/ContentFilter.swift

```diff
@@ -31,13 +31,21 @@ class ContentFilter: FileFilter {
     /// - returns: `true` if the
     func filter() -> Bool {
         // Use simple string matching first since it's more performant.
-        if !content.contains("Component") {
+        if !content.contains("Component") && !content.contains("Dependency") {
             return false
         }
 
         // Match actual component inheritance using Regex.
         let containsComponentInheritance = (Regex(": *Component *<").firstMatch(in: content) != nil)
-        return containsComponentInheritance
+        if containsComponentInheritance {
```

Comment:
Can we please, I think its really redundant to have these if:

return containsComponentInheritance || containsDependencyInheritance

Reply:
Yah so i was going to originally say this:
return (Regex(": *Component *<").firstMatch(in: content) != nil) || (Regex(": *Dependency").firstMatch(in: content) != nil)
Itll short circuit and be the same efficiency! but I have a feeling you are going to say this not readable...

Reply:
That's definitely not readable haha. Let's not do that.